### PR TITLE
Spot It MVP

### PR DESCRIPTION
**Pull request by Tanvez**
- [x] Connect database to spot-it views and other files
- [x] Working game with placeholder images
- [ ] Fix misc bugs?
- [x] UI Rotate and resize images similar to spot it card game
- [x] Update CSS to highlight selected images

### CODE REVIEW

File: parser_tool/static/js/src/spot_it.js [hunk 1]
```diff
@@ -0,0 +1,482 @@
+(function ($) {
+  /**
+   * Theorem that given a prime number you could generate N * N + N + 1 cards with N + 1 symbols
+   * whose relationship between each card has a maximum of one related symbol
+   * Note: reason we use prime numbers is due to the theorem which uses a diagonalization method for generating the 1 to 1 relationship between the cards
+   * Ref: https://math.stackexchange.com/questions/1303497/what-is-the-algorithm-to-generate-the-cards-in-the-game-dobble-known-as-spo
+   **/
+  const generateDeckFromPrimeNum = (n) => {
```
Comment (arthurian) [inline parser_tool/static/js/src/spot_it.js:8]: If I'm understanding correctly, there are 3 variables that are determined by the prime number passed to this function:

```
C = total number of cards in the deck
S = total number of symbols (or images)
N = number of symbols per card
```

So given a prime number `n`, we get `n+1` symbols per card, with `n^2 + n + 1` total cards in the deck and `n^2 + n + 1` total symbols. And the output of this function is an array of cards with the symbols on each, as represented by a number from `0...N`. Is that correct?

File: parser_tool/static/js/src/spot_it.js [hunk 2]
```diff
@@ -0,0 +1,482 @@
...
+    let card2 = [];
+    let centralCard = [];
+    let deckElementP2 = "";
+    state.p1Score = 0;
+    state.p2Score = 0;
+    const data = JSON.parse(document.getElementById("data").textContent);
+    // static numbers to validate against data length
+    // image/button element position:[right,top]
+    // TODO make more dynamic - using static positions for images so they do not overlap
+    const maxNumReqImages = [
```
Comment (arthurian) [inline parser_tool/static/js/src/spot_it.js:116]: Nice declarative structure for the possible game configurations (static is fine for now).

**Review by arthurian — APPROVED** [review]
Great work on the MVP @Tanvez!

File: parser_tool/forms.py [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+
+        return [s.strip() for s in value.split("\n") if s !='']
+
+    def validate(self, value):
+        """Check if value contains MIN words."""
+        super().validate(value)
+        # Check if words in the database
+        value = [w for w in value if w != '']
+        qs_lemmas = models.Lemma.objects.filter(lemma__in=value).distinct().values_list('lemma', flat=True)
+        print("WordListField validate: ", qs_lemmas, len(qs_lemmas))
```
[the remark this answers is not in the document]
Comment (Tanvez) [reply parser_tool/forms.py:34]: Cool! Will update that to use Logging.

[the remark this answers is not in the document]
Comment (Tanvez) [reply parser_tool/forms.py:34]: Should we output the logging into a separate file? or log to the console?

File: parser_tool/static/js/src/spot_it.js (hunk 1 above)
Comment (Tanvez) [reply parser_tool/static/js/src/spot_it.js:8]: Yes! Thats right! Something interesting is the pattern behind the algorithm. I still dont understand it 100%. I found a very cool visualization off of reddit below.
spoti-it-reddit
.

Comment (arthurian) [reply parser_tool/static/js/src/spot_it.js:8]: That's really neat! I'm going to have to study that a bit to make sense of it, but a very cool visualization.

**Pull request merged**